storagemarket/deal_commp.go

```diff
@@ -196,7 +196,8 @@ func getCarSize(filepath string, rd *carv2.Reader) (int64, error) {
 	var size int64
 	switch rd.Version {
 	case 2:
-		size = int64(rd.Header.DataSize)
+		padding := rd.Header.DataOffset - carv2.PragmaSize - carv2.HeaderSize
```

Comment:
I think we should probably check that `padding` > 0, and log an error if this is not the case, including the `carv2.PragmaSize` and `carv2.HeaderSize` if they are malformed.

Reply:
I think having a padding is normal but it is optional so having a 0 padding means that the size on the header is correct. 

We can log info if the padding > 0, but i don’t think its an error.

Reply:
I am worried about the malformed case, where padding ends up being negative, and therefore we calculate `size` to be smaller than it actually is.

Reply:
ah got it. If it's negative value then we can add a error log to it i.e if `padding < 0`, we log error.

Reply:
```
padding := rd.Header.DataOffset - carv2.PragmaSize - carv2.HeaderSize
if padding < 0 {
	return 0, fmt.Errorf("CARv2 file is invalid: padding is negative")
}
```